Source/DataModels/Windows/SubWindows/ComponentWindows/WindowComponentScript.cpp

```diff
@@ -99,7 +100,7 @@ void WindowComponentScript::DrawWindowContents()
 				
 					label = stringField.name;
 					finalLabel = label + separator + thisID;
-					if (ImGui::InputText(finalLabel.c_str(), (char*)(value.c_str()), 24))
+					if (ImGui::InputText(finalLabel.c_str(), const_cast<char*>(value.c_str()), 24))
```

Comment:
`const_cast` are generally avoidable (and it's a bad sign if they can't be avoided), I think this can be replaced with
```suggestion
					if (ImGui::InputText(finalLabel.c_str(), value.data(), 24))
```
Though this is only available since C++17. In that version, `data` returns a non-const `char*` for a non-const `std::string`, whereas in previous versions it always returned a `const char*`. More information in [the documentation](https://en.cppreference.com/w/cpp/string/basic_string/data).

Reply:
Felt wrong doing a `const_cast` though, thanks for pointing out a better solution!